src/components/bcros/filing/common/HeaderActions.vue

```diff
@@ -41,24 +93,97 @@
 
 <script setup lang="ts">
 import { FilingTypes } from '@bcrs-shared-components/enums'
+import { z } from 'zod'
 import { type ApiResponseFilingI, FilingStatusE, isFilingStatus, isStaffFiling } from '#imports'
+import { FilingCorrectionTypesE } from '~/enums/filing-correction-types-e'
 
 const { getStoredFlag } = useBcrosLaunchdarkly()
 const { hasRoleStaff } = storeToRefs(useBcrosKeycloak())
 const { isAllowedToFile, isBaseCompany, isDisableNonBenCorps, isEntityCoop, isEntityFirm } = useBcrosBusiness()
 const { currentBusiness } = storeToRefs(useBcrosBusiness())
 const { isBootstrapFiling } = useBcrosBusinessBootstrap()
+const { goToEditPage } = useBcrosNavigate()
 
 const isCommentOpen = ref(false)
-
+const filings = useBcrosFilings()
 const isExpanded = defineModel('isExpanded', { type: Boolean, required: true })
+const showFilingModal = ref(false)
 
 const filing = defineModel('filing', { type: Object as PropType<ApiResponseFilingI>, required: true })
 
 const t = useNuxtApp().$i18n.t
 
 const isTypeStaff = computed(() => isStaffFiling(filing.value))
 
+const setShowFilingModal = (value: boolean) => {
+  showFilingModal.value = value
+}
+
+const formState = ref({
+  correctionType: null
+})
+const correctionForm = ref()
+const submissionInProgress = ref(false)
+const filingError = ref('')
+const correctionTypes = [
+  { value: FilingCorrectionTypesE.CLIENT, label: t('button.filing.correction.client') },
+  { value: FilingCorrectionTypesE.STAFF, label: t('button.filing.correction.staff') }
+]
+const correctionFormSubmit = async function () {
+  submissionInProgress.value = true
+  const valid = await correctionForm.value.validate(null, { silent: true })
+  if (valid === false) {
+    return
+  }
+
+  const correctionType = formState.value.correctionType
+
+  const response = await filings.createFiling(
+    currentBusiness.value,
+    FilingTypes.CORRECTION,
+    {
+      comment: '',
+      correctedFilingDate: dateToYyyyMmDd(new Date(filing.value.submittedDate)),
+      correctedFilingId: filing.value.filingId,
+      correctedFilingType: filing.value.name,
+      type: correctionType
+    }
+  )
+  submissionInProgress.value = false
+  if (response.error?.value) {
+    console.error(response.error.value)
+    filingError.value = response.error.value
+    return
+  }
+  filingError.value = ''
+  const draftFilingId = '' + response.header?.filingId
+  if (!draftFilingId) {
+    filingError.value = 'Unable to get correction filing id'
+    return
+  }
+  const path = `/${currentBusiness.value.identifier}/correction/`
+  const params = { 'correction-id': draftFilingId }
+  goToEditPage(path, params)
```

Comment:
This doesn't seem to work, ie:

![image](https://github.com/user-attachments/assets/c3add9e0-8895-4c73-a9ed-362df745217f)

The correction-id is null.

Reply:
My new PR fixed this thanks for the catch

Reply:
Your fix works fine. Please resolve this conversation (as I cannot). Thanks.